packages/deployments/contracts/contracts_forge/messaging/connectors/wormhole/WormholeSpokeConnector.t.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {WormholeSpokeConnector} from "../../../../contracts/messaging/connectors/wormhole/WormholeSpokeConnector.sol";
+import {BaseWormhole} from "../../../../contracts/messaging/connectors/wormhole/BaseWormhole.sol";
+import "../../../../contracts/messaging/interfaces/ambs/wormhole/IWormholeRelayer.sol";
+import {MerkleTreeManager} from "../../../../contracts/messaging/MerkleTreeManager.sol";
+
+import "../../../utils/ConnectorHelper.sol";
+import "../../../utils/Mock.sol";
+
+contract WormholeSpokeConnectorTest is ConnectorHelper {
+  using stdStorage for StdStorage;
+
+  // ============ Storage ============
+  uint16 _chainIdL1 = 2;
+  uint16 _chainIdL2 = 4;
+  uint256 _gasCapL1 = 300000;
+  uint256 _gasCapL2 = 300000;
+
+  // ============ Test set up ============
+  function setUp() public {
+    vm.clearMockedCalls();
+
+    // Allow future contract mock
+    vm.etch(_amb, new bytes(0x42));
+    vm.etch(_rootManager, new bytes(0x42));
+
+    // Get the n+1 deployment address
+    _l1Connector = payable(address(bytes20(keccak256("_l1Connector"))));
+    _merkle = address(new MerkleTreeManager());
+
+    // Deploy
+    vm.prank(_owner);
+    _l2Connector = payable(
+      address(
+        new WormholeSpokeConnector(
+          _l1Domain,
+          _l2Domain,
+          _amb,
+          _rootManager,
+          _l1Connector,
+          _processGas,
+          _reserveGas,
+          0, // uint256 _delayBlocks
+          _merkle,
+          address(1), // watcher manager
+          _gasCapL1,
+          _chainIdL1
+        )
+      )
+    );
+  }
+
+  // ============ Utils ============
+
+  // ============ send ============
+
+  // Happy path L1
+  function test_WormholeSpokeConnector_send_sendAndEmitEvent() public {
```

Comment:
fuzz data? not sure why it has to be an actual merkle root from a tree vs random bytes32